### PR TITLE
Feat account creation

### LINKED ISSUE #10: Create code samples demonstrating account creation on the wiki via API for different cases
The account creation via the API is a complex process. It works differently in different cases: when creating an account for someone else, on a wiki with or without special authentication extensions installed. It might be good to develop code samples for different use cases and embed them here:  https://www.mediawiki.org/wiki/API:Account_creation. Three use cases to develop code samples for:
* A simple example, the case of a wiki where there aren't any special authentication extensions installed
* Complex example, the case of a wiki with special authentication extensions installed 
 * On a wiki with captcha enabled (Extension:ConfirmEdit)
 * On a wiki with two factor authentication enabled (Extension:OpenID or OATHAuth)

### PR DESCRIPTION
**Pull request by ARsquare**
Fixes #10 
As discussed in the comment :
https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples/issues/10#issuecomment-469930849

I've made the said changes. Please have a look @srish

### CODE REVIEW

**Comment by srish** [comment]
@ARsquare Thanks a lot for your work on this! :) I will take a look at it soon..

File: python/create_account_with_captcha.py [hunk 1]
```diff
@@ -18,8 +18,11 @@
 from flask import Flask, render_template, flash, request
 
 S = requests.Session()
-WIKI_URL = "https://test.wikipedia.org"
-API_ENDPOINT = WIKI_URL + "/w/api.php"
+TEST_WIKI_URL = "https://test.wikipedia.org"
```
Comment (srish) [inline python/create_account_with_captcha.py]: We are declaring endpoints twice now. You can keep this as it was initially `WIKI_URL` & `API_Endpoint.`

File: python/create_account_with_captcha.py [hunk 2]
```diff
@@ -30,36 +33,48 @@
...
+            'confirm_password': request.form['retype'],
             'email': request.form['email'],
             'captcha_word': request.form['captcha-word'],
-            'captcha_id': captcha_fields['captchaId']['value']
-        }
+            'captcha_id': captcha_fields['captchaId']['value']}
 
         create_account(details)
 
+    register_fields = list()
```
Comment (srish) [inline python/create_account_with_captcha.py]: In place of `list()` you can just use `[]`.

File: python/create_account_with_captcha.py [hunk 3]
```diff
@@ -30,36 +33,48 @@
...
+    registration_fields = get_registration_fields()
```
Comment (srish) [inline python/create_account_with_captcha.py]: Variable names could be improved in this section..

File: python/create_account_with_captcha.py [hunk 4]
```diff
@@ -75,13 +90,42 @@ def get_captcha_fields():
...
+            'meta': 'authmanagerinfo',
+            'amirequestsfor': 'create',
+            'format': 'json'})
+
+    data = response.json()
+    query = data and data['query']
+    authmanagerinfo = query and query['authmanagerinfo']
+    fields = authmanagerinfo and authmanagerinfo['requests']
+
+    ret_field = dict()
```
Comment (srish) [inline python/create_account_with_captcha.py]: In place of `dict()` just use `{}`

Comment (srish) [reply python/create_account_with_captcha.py]: Also, `ret_field` name could be more readable. May be use fields instead. And, then change the name of fields variable to `requests`.

File: python/create_account_with_captcha.py [hunk 5]
```diff
@@ -75,13 +90,42 @@ def get_captcha_fields():
...
+
+    for k in fields:
```
Comment (srish) [inline python/create_account_with_captcha.py]: Variable names in this for loop could be slightly improved and made more self-explanatory. We don't want them to be lengthy either..

File: python/create_account_with_captcha.py [hunk 6]
```diff
@@ -30,36 +33,48 @@
...
             'captcha_word': request.form['captcha-word'],
-            'captcha_id': captcha_fields['captchaId']['value']
-        }
+            'captcha_id': captcha_fields['captchaId']['value']}
 
         create_account(details)
 
+    register_fields = list()
+    registration_fields = get_registration_fields()
+    for x in registration_fields:
```
Comment (srish) [inline python/create_account_with_captcha.py]: Variable names in this for loop could be slightly improved and made more self-explanatory. We don't want them to be lengthy either..

File: python/create_account_with_captcha.py [hunk 7]
```diff
@@ -75,13 +90,42 @@ def get_captcha_fields():
...
+
+    ret_field = dict()
+
+    for k in fields:
+        if k['id'] == 'MediaWiki\\Auth\\PasswordAuthenticationRequest' or k['id'] == 'MediaWiki\\Auth\\UserDataAuthenticationRequest':
+            k_field = k and k['fields']
+            for x in k_field:
+                ret_field[x] = k_field[x]
+
+    if not ret_field:
```
Comment (srish) [inline python/create_account_with_captcha.py]: You can get rid of this if block..

File: python/create_account_with_captcha.py (hunk 3 above)
Comment (ARsquare) [reply python/create_account_with_captcha.py]: Done!

File: python/create_account_with_captcha.py (hunk 1 above)
Comment (ARsquare) [reply python/create_account_with_captcha.py]: Actually, those were meant to be two different endpoints.

**Comment by srish** [comment]
@ARSquare I merged this pull request by mistake even though I had requested changes on it :-/ If you have time maybe take a look at the requested changes in https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples/pull/100#pullrequestreview-217481364 and create another pull request for that..

**Pull request merged**